subsys/testsuite/ztest/src/ztest.c

```diff
@@ -79,6 +79,7 @@ static int cleanup_test(struct unit_test *test)
 		PRINT("Test %s failed: Unused mock return values\n",
 		      test->name);
 		ret = TC_FAIL;
+	} else {
 	}
 
```

Comment:
This test could use some improvement. Perhaps I can do it here as part of refactoring approach to this patch for if...else if constructs. 

I do not understand why check for `!ret` like this with ret implementation. -- ! is a logical operator. !ret has value 0 if ret has value non-zero. The value is initialized as `int ret = TC_PASS` (non-zero). Would it be more informative to initialize as `ret=0`, then this check makes?

Reply:
Oh, `TC_PASS` expands to 0.